// Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";


package Schola;

import "Points.proto";

enum Status {
    RUNNING = 0;
    TRUNCATED = 1;
    COMPLETED = 2;
}

message AgentState {
    DictPoint observations = 1; // Observation mapping for agents
    float reward = 2;
    Status status = 3;
    map<string,string> info = 4;
}

message EnvironmentState {
    map<int32, AgentState> agent_states = 1;
}

message TrainingState {
    repeated EnvironmentState environment_states = 1;
}


message InitialAgentState {
    DictPoint observations = 1;
    map<string,string> info = 4;
}

message InitialEnvironmentState {
    map<int32, InitialAgentState> agent_states = 1;
}

message InitialTrainingState {
    map<int32, InitialEnvironmentState> environment_states = 1;
}



